go.mod

```diff
@@ -100,6 +100,7 @@ require (
 	sigs.k8s.io/cluster-api-provider-openstack v0.3.2 // indirect
 	sigs.k8s.io/controller-runtime v0.6.3
 	sigs.k8s.io/controller-tools v0.3.0
```

Comment:
run `make vendor`

Reply:
Yep, just pushed it a second ago.

Reply:
Hang on. I did `go mod vendor`. `make vendor` does something else. Let me double check.

Reply:
@mjudeikis so `make vendor` updates vendor. I re-porpused this PR to update vendor (had to fix some incompatibility with `azure-sdk-for-go v49.0.0`) and submitted a separate PR just for docs: https://github.com/Azure/ARO-RP/pull/1214

Not sure if we actually want to update dependencies or just `go mod tidy` and `go mod vendor` to fix missing yaml package.